### ISSUE TITLE
Add a rule to report direct Bun.env reads

### ISSUE THREAD

**Issue by dBianchii** [biomejs/biome#10617]
### Problem

Projects that centralize environment access currently cannot ban direct `Bun.env` reads with Biome config without also banning the whole `Bun` global.

For example, this currently passes unless a project has a separate custom/non-Biome check:

```ts
const fromBun = Bun.env["NODE_ENV"];

void fromBun;
```

`noRestrictedGlobals` cannot express this precisely because it targets global identifiers. Adding `"Bun.env"` to `deniedGlobals` does not report `Bun.env`, while adding `"Bun"` would also ban valid Bun APIs such as `Bun.file`, `Bun.serve`, `Bun.$`, and `Bun.write`.

### Requested solution

Add a lint rule, or configurable rule support, that can report direct `Bun.env` reads.

This does not need to be `style/noProcessEnv` if maintainers want that rule to stay scoped to `process.env`. A separate rule such as `noBunEnv`, or a more general runtime env access rule, would solve the consumer need.

### Related

This was originally discussed alongside `process.env` aliases in #10447. A maintainer noted that `Bun.env` may belong in a different rule because `noProcessEnv` is meant to catch `process.env`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
